lexer.go

```diff
@@ -59,8 +59,14 @@ type Scanner struct {
 
 	// lastKeyword records the previous keyword returned by scan().
 	// determine whether an optimizer hint should be parsed or ignored.
-	lastKeyword               int
-	keywoardBeforeLastKeyword int
+	lastKeyword int
+	// lastKeyword2 records the keyword before lastKeyword, it is used
+	// to disambiguate hint after for update, which should be ignored.
```

Comment:
please add a test case for `select ... for update /*+ should ignore this */` then.

Reply:
> please add a test case for `select ... for update /*+ should ignore this */` then.

I have added it in PR https://github.com/pingcap/parser/pull/1004, see https://github.com/pingcap/parser/pull/1004/files#diff-eff279f70f1a81e2abdae4e57736a72120ce3a184bc25ccdbbb60cb10b3027c5R3040-R3044

Reply:
okay